src/internal/scheduled/scheduleIterable.ts

```diff
@@ -1,45 +1,49 @@
+/** @prettier */
 import { Observable } from '../Observable';
 import { SchedulerLike } from '../types';
 import { Subscription } from '../Subscription';
 import { iterator as Symbol_iterator } from '../symbol/iterator';
+import { isFunction } from '../util/isFunction';
 
 export function scheduleIterable<T>(input: Iterable<T>, scheduler: SchedulerLike) {
   if (!input) {
     throw new Error('Iterable cannot be null');
   }
-  return new Observable<T>(subscriber => {
+  return new Observable<T>((subscriber) => {
     const sub = new Subscription();
     let iterator: Iterator<T>;
     sub.add(() => {
       // Finalize generators
-      if (iterator && typeof iterator.return === 'function') {
-        iterator.return();
-      }
+      isFunction(iterator.return) && iterator.return();
```

Comment:
issue: The replaced code tests `iterator`, so optional chaining should be used, IMO.
```
isFunction(iterator?.return)
```